mysql-test/mysql-test-run.pl

```diff
@@ -1719,17 +1720,28 @@ sub command_line_setup {
     $opt_valgrind= 1;
     $opt_valgrind_mysqld= 1;
 
+    my @valgrind_args_orig= @valgrind_args;
+
```

Comment:
Can't understand the purpose of @valgrind_args_orig. Switching the order of push and unshift should do the trick just as well, with the original @valgrind_args? 

```
push(@valgrind_args, "--trace-children=yes") 
  unless @valgrind_args;
unshift(@valgrind_args, "--tool=callgrind"); 
```

(or maybe rather ```push(@valgrind_args, "--tool=callgrind")```? Why do we need it at the beginning?)